pythonforandroid/pythonpackage.py

```diff
@@ -0,0 +1,684 @@
+    # these issues.
+
+    clean_up_path = False
+    path_type = "source"
+    path = parse_as_folder_reference(package)
+    if path is None:
+        # This is not a path. Download it:
+        (path_type, path) = get_package_as_folder(package)
+        if path_type is None:
+            # Download failed.
+            raise ValueError(
+                "cannot get info for this package, " +
+                "pip says it has no downloads (conditional dependency?)"
+            )
+        clean_up_path = True
+
+    try:
+        build_requires = []
+        metadata_path = None
+        if path_type != "wheel":
+            # We need to process this first to get the metadata.
+
+            # Ensure pyproject.toml is available (pep517 expects it)
+            if not os.path.exists(os.path.join(path, "pyproject.toml")):
+                with open(os.path.join(path, "pyproject.toml"), "w") as f:
+                    f.write(textwrap.dedent(u"""\
+                    [build-system]
+                    requires = ["setuptools", "wheel"]
+                    build-backend = "setuptools.build_meta"
+                    """))
+
+            # Copy the pyproject.toml:
+            shutil.copyfile(
+                os.path.join(path, 'pyproject.toml'),
+                os.path.join(output_path, 'pyproject.toml')
+            )
+
+            # Get build backend from pyproject.toml:
+            with open(os.path.join(path, 'pyproject.toml')) as f:
+                build_sys = pytoml.load(f)['build-system']
+                backend = build_sys["build-backend"]
+
+            # Get a virtualenv with build requirements and get all metadata:
+            env = BuildEnvironment()
+            metadata = None
+            with env:
+                hooks = Pep517HookCaller(path, backend)
+                env.pip_install([transform_dep_for_pip(req) for req in build_requires])
+                reqs = hooks.get_requires_for_build_wheel({})
+                env.pip_install([transform_dep_for_pip(req) for req in reqs])
+                try:
+                    metadata = hooks.prepare_metadata_for_build_wheel(path)
+                except Exception:  # sadly, pep517 has no good error here
+                    pass
+            if metadata is not None:
+                metadata_path = os.path.join(
+                    path, metadata, "METADATA"
+                )
+        else:
+            # This is a wheel, so metadata should be in *.dist-info folder:
+            metadata_path = os.path.join(
+                path,
+                [f for f in os.listdir(path) if f.endswith(".dist-info")][0],
+                "METADATA"
+            )
+
+        # Copy the metadata file:
+        shutil.copyfile(metadata_path, os.path.join(output_path, "METADATA"))
+    finally:
+        if clean_up_path:
+            shutil.rmtree(path)
+
+
+def is_filesystem_path(dep):
+    if dep.startswith("/") or dep.startswith("file://") or (
```

Comment:
same minor thing:
```python
dep.startswith(("/", "file://"))
```

Reply:
I feel pretty dumb for not knowing that pattern yet :joy: